### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by janwilmans**
After getting the question: so what is libguarded useful for, reading the github description it looks like a wrapper for std::mutex, but I dont get the point. I thought I would take a stab at a more to the point description. I hope you like it :)

### CODE REVIEW

**Comment by Andres-Hernandez** [comment]
You might consider adding links to the CppCon 2017 presentation. After watching just a few minutes of it, it becomes clear why the library is a great. I mean, I can write correct code with raw pointers, but for almost all cases why would I want to? Similarly, I can write correct code using raw threads and mutexes, but after watching the videos, why would I want to?

https://www.youtube.com/watch?v=GNw3RXr-VJk
https://www.youtube.com/watch?v=sDLQWivf1-I&t=1984s

**Comment by agserm** [comment]
Thanks for the pull request and the feedback! Will look at merging this sometime next week.

File: src/libguarded/guarded.hpp [hunk 1]
```diff
@@ -19,6 +19,48 @@
 namespace libguarded
 {
 
+template <typename T, typename D>
+struct Handle
+{
+    Handle(T* t, D& d) :
+        ref(*t),
```
Comment (ericLemanissier) [inline src/libguarded/guarded.hpp]: by dereferencing t, you hit undefined behaviour if t is nullptr. the ref member variable should be stored as a pointer, not a reference

File: src/libguarded/guarded.hpp [hunk 2]
```diff
@@ -19,6 +19,48 @@
...
+    }
+
+    T* operator->() const
+    {
+        return &ref;
+    }
+
+    const T& operator*() const
+    {
+        return *ref;
```
Comment (ericLemanissier) [inline src/libguarded/guarded.hpp]: there should be no dereferencing here: you are dereferencing T, not T*. The body should be identical to the non const overload below

**Comment by agserm** [comment]
Everything here appears to be resolved. In the future, please open a new PR for each item.

**Pull request closed without merging**